src/couch_mrview/src/couch_mrview_util.erl

```diff
@@ -525,6 +527,39 @@ validate_args(#mrst{} = State, Args0) ->
     end.
 
 
+apply_limit(ViewPartitioned, Args) ->
+    MaxLimitType = case ViewPartitioned of
+        true -> "partition_";
+        false -> ""
+    end,
+    
+    MaxLimit = case config:get("query_server_config",
```

Comment:
Using `config:get_integer("query_server_config", MaxLimit, ?MAX_VIEW_LIMIT)` would clean up the logic and avoid having to do the whole infinity dance.

Reply:
From what I can see `config:get_integer` cannot handle `infinity` as an input.